### DISCUSSION TITLE
How to re-enter a running/suspended workflow built with 3.x Studio?

### DISCUSSION THREAD

**Discussion by doubleG7** [elsa-workflows/elsa-core#5654] [Q&A]
The workflow begins with an HTTPEndpoint, accepting a body and several parameters. Further along, there's a join activity that waits for additional input before the workflow can finish. How can the suspended workflow be re-invoked with a parameter? Should this be done using the definitionId or the workflowInstanceId with the execute API? Many thanks!

**Comment by glime-ai** [comment]
To re-invoke a suspended workflow with additional parameters, you should use the workflow instance ID, not the definition ID. The workflow instance ID uniquely identifies the running instance of the workflow that has been suspended and is waiting for additional input.

You can use the `/workflow-instances/{workflow_instance_id}/dispatch` endpoint to asynchronously execute the workflow instance with the additional parameters. This endpoint will enqueue the workflow for execution and return immediately, allowing the workflow to resume with the provided input.

Here's an example of how you might call this endpoint using `curl`:

```bash
curl --location --request POST 'https://your-elsa-server/workflows/instances/{workflow_instance_id}/dispatch' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer {your_api_key}' \
--data-raw '{
    "input": {
        "ParameterName": "ParameterValue"
    }
}'
```

Replace `{workflow_instance_id}` with the actual ID of the workflow instance you want to resume, `{your_api_key}` with your actual API key, and adjust the `input` object to include the parameters you want to pass to the workflow.

Make sure that the workflow is designed to accept and handle the input parameters you are passing. If the workflow is waiting for a signal, you may need to use the Signal Received activity to resume the workflow, as mentioned in the context sources.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
